### PR TITLE
Vegetable Audit

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
None

#### Purpose of change
I said I'd be back for the rest of comestibles.

#### Describe the solution
- Audited most of calories, vitamin contents, pre-apo prices, volumes and weights for our common vegetables. I left the wilderness bullshit alone for now because I have no idea where to find data on most of it
- Cabbage and lettuce were renamed to be handfuls of shredded versions of themselves. Cabbage and lettuce heads have been added as the grown crop instead. Those need to be disassembled into edible forms. Left the heads edible for cattle mutants because fuck it why not.
- Renamed spinach, bamboo shoots and beans to a handfuls
- Edited the descriptions for chicory-related foodies so they don't lie
- Made eggplant big boye - unusable in recipes now, needs to be disassembled, updated the description to reflect that
- Lowered the amount of sugar obtained from beets to match weight decrease
- Renamed cucumbers to salad cucumbers. Added gherkins (the small ones used for pickles) as a new crop entirely. Haven't made gherkins spawn anywhere as an item form because frankly I've never seen them be sold anywhere unpickled. The seeds for them are still in the generic seed spawngroup, though, so those should spawn in the world.
- Removed raw lentils because they were entirely unused throughout the repo
- Calculated the max length available for plastic tupperware from pythagorean shit so that carrots can fit in there

Other items audited to please the tests:
- canned tomatoes - values taken from crushed canned tomatoes
- tomato juice
- tomato soup
- ketchup
- tomato sauce
- V8 - values taken from organic variety

#### Additional context
https://www.eatthismuch.com/food/browse/?q=&type=food - for most of the food in question. Both for prices, caloric content, vitamins and sizes. There are, however, exceptions
- the volume of plantain has been calculated proportionally from the volume of a banana
- the weight and volume of horseradish root has been unchanged, instead I audited the nutritional value based on the weight of our item
- celery and carrot specifically had their volume calculated as cylinders of 7.25" length and 1.25" diameter
- cucumber had its volume calculated as a cylinder of 8.25" length and 1.5" diameter
- lettuce and cabbage are based off 13 cups of shredded cabbage and 7 cups of shredded ice lettuce - those are rounded weight equivalents to medium heads
- bottle gourd data sourced from https://www.nutritionix.com/food/bottle-gourd - proportionally from the weight, with the assumption that 50% of the bottle gourd weight is actually edible because I couldn't find the actual percentage online

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
My advice here is to make the current lettuce and cabbage items stay the same, and make a new lettuce/cabbage head which must be disassembled before eating, like a melon. After that we just move the seed data and some of the spawns over to the heads.

Grocery stores often sell pre-chopped lettuce and cabbage, and people do store them that way at home, so it won't be too strange.

**Comment by Karol1223** [comment]
> My advice here is to make the current lettuce and cabbage items stay the same, and make a new lettuce/cabbage head which must be disassembled before eating, like a melon (probably shouldn't need a knife though). After that we just move the seed data and some of the spawns over to the heads.

Yeah the plan was to do it like with a melon, but back when I was doing the melons I did it the way I started it now - replacing old items this way. You're right in that it'd probably be less of a hassle to make a new item rather than manually swapping IDs everywhere, though. 

> Grocery stores often sell pre-chopped lettuce and cabbage, and people do store them that way at home, so it won't be too strange.

...they do? I mean, I don't know how it is in the US, but I have never seen pre-chopped lettuce/salad be sold around here. I've always seen them come in full heads...

**Comment by fairyarmadillo** [comment]
> ...they do? I mean, I don't know how it is in the US, but I have never seen pre-chopped lettuce/salad be sold around here. I've always seen them come in full heads...

Yep!

image

There are also often open bins full of different kinds of pre-chopped lettuce, like a bulk kind of thing, and you can take as much as you like. This is all sold alongside whole heads.

image

You can get 'em in these plastic tubs, too.

**Comment by Karol1223** [comment]
Ah, okay, I've seen the ones in the little plastic boxes - I forgot those exist, frankly. I usually ignore their existence because they're out of the way in any store I've been to lmao. This kind of reminds me how bland our ingame stores are, but that's a separate problem I suppose. I'll try to handle that either tomorrow or... well, sometimes next week.

**Comment by Karol1223** [comment]
@fairyarmadillo May I ask where you got the values for pumpkins in #65924? I've been trying to find anything for those and I'm completely stumped. I have kcal and vitamin/g but I have no sources for average weights/volumes...

**Comment by fairyarmadillo** [comment]
> @fairyarmadillo May I ask where you got the values for pumpkins in #65924? I've been trying to find anything for those and I'm completely stumped. I have kcal and vitamin/g but I have no sources for average weights/volumes...

I based them off of sugar pie pumpkins, which are the variety most commonly sold as eating pumpkins. Carving pumpkins are technically edible, but don't taste as good and have much less flesh and much more water by weight, so most of them work out to be roughly the same once prepared and cooked, except for the largest varieties (which I've never seen in a store).

One problem is that the nutrition facts you find online are almost always for cooked pumpkin.

https://nesfp.nutrition.tufts.edu/world-peas-food-hub/world-peas-csa/produce-recipes/sugar-pumpkin

This site lists the pumpkin as between 4 and 8 pounds. In my own experience, 8 is an overestimation. Other sources say 3-6. I settled on 5 pounds, which is on the largeish side that I've encountered when growing or buying these pumpkins myself.

https://ohsheglows.com/how-to-roast-a-sugar-pumpkin-make-fresh-pumpkin-puree-a-step-by-step-photo-tutorial/

This site says you get about 4 cups of cooked pumpkin out of a 4-5 pound pumpkin. That seems a little high to me, but it's a nice even number for our purposes, and the calorie values are low enough that I don't really think it matters too much.

https://stopandshop.com/groceries/produce/fresh-vegetables/squash/fresh-pumpkins/pumpkins-sugar-pie-for-baking-1-ea.html

This site lists 30kcal per cup of cooked pumpkin. If there are 4 of those in one pumpkin, the total works out to be 120kcal. This matches what we have in the game.

image

I frequently cook with these pumpkins IRL. The yield in seeds is actually pretty disappointing - this is what I got out of my last one. In the spirit of generosity I made the total seed amount roughly 1 cup, and I guess at the outside that's how much you might get once you pick all the pulp out. Eating an entire cup of pumpkin seeds is preposterous as they're fairly high calorie, so that's split into six little 60 kcal piles. Keep in mind that that's 1 cup with the shells on. They're like sunflower seeds, the actual seed is much smaller inside.

**Comment by fairyarmadillo** [comment]
I actually suspect that the current kcal values for pumpkin seeds are for seeds with the shells already removed, but our current pumpkin seed volume is based on seeds with the shells still on. You probably only get like 1/4 to 1/2 cup of those from a whole pumpkin.

**Comment by Montimorency** [comment]
HEY!!!

@Karol1223 
@fairyarmadillo 

### I was looking into setting up a fork and PR for the following. Could you please add it here instead? 

In-game, a 5 lb pumpkin contains 90 calories, for 4 slices of 22 calories each. Obviously this is far too low. For cites, the links indicate that "[a] 5 pound fresh pumpkin will make 4 to  4½ cups of cooked puree or mashed pulp", and "[t]here are 49 calories in a cup of cooked pumpkin."

Zucchini in the game, weight for weight, provide nearly 10 times as many calories as pumpkin, whereas "[i]n comparison to zucchini, a similar fruit, pumpkin contains slightly more than double the calories per cup." The USDA link agrees with this and is probably the original source. Indeed, whoever programmed pumpkin nutrition in-game seems almost to have applied 100 g nutrition - 20 kcal - for 1 lb of pumpkin in-game. 

So I'm adjusting whole pumpkin calories from 90 to 200, and cut pumpkin from 22 to 49. As for vitamins, for some reason cut pumpkin in-game overwhelmingly loses its Calcium and Iron , but not Vitamin C, content, and this should be evened out for whole pumpkin in addition to changes in overall content. (The same applies for some other disassembly produce, such as watermelon, but one thing at a time.)

With USDA figures the whole pumpkin would thus go from 

83 mg Ca/3 mg Fe/62 mg VIT C 
to 
150 mg Ca/6 mg Fe/50 mg VIT C. 

Cut pumpkin should go from 

10 mg Ca/564 µg Fe/15 mg VIT C 
to 
37 mg Ca/1.4 mg Fe/11.5 mg VIT C, 

representing a single-digit pp loss of vitamins (and calories) in cutting. 

Note that I am not representing any potential differences here between real world cooked/mashed pumpkin and the game item "cut pumpkin" for simplicity's sake, I believe in accordance with how nutrition is usually treated in the current model. I can also conserve nutrition 1:1 between whole pumpkin and sub-pumpkin if that is preferred. 

https://www.howmuchisin.com/produce_converters/pumpkin
https://www.verywellfit.com/pumpkin-nutrition-facts-calories-carbs-and-health-benefits-4165492
https://fdc.nal.usda.gov/fdc-app.html#/food-details/168449/nutrients

**Comment by fairyarmadillo** [comment]
If you're working on your own fork, that's nothing to do with me, but if you just mean a PR, that can probably be handled by Karol here.

I provided a site that says 1 cup is 30kcal, you pointed out one that says 49kcal. Libby's canned pumpkin is 45kcal per cup - maybe that should be our guidepost, as that's regulated by the USDA.

And yeah realistically you'd get less than 4 cups of cooked pumpkin out of a 4 cup pumpkin, but there's a necessary level of abstraction going on, but not all fruit is the same size IRL.

**Comment by Montimorency** [comment]
I haven't done anything with Github, I was just looking into the process and preparing the files. It would make the most sense simply to fold the changes right into this PR. 

To be exact, the USDA page - AFAIK USDA is the gold standard reference that C:DDA uses for food items - ascribes 49 calories to 245 g of mashed pumpkin, whereas cut pumpkin in-game is 525g. Maybe it includes some of the rind. As you say, I prefer to just abstract it and force an agreement with 5 lb pumpkin = 4 cut pumpkin = 4 cups mashed.

**Comment by Montimorency** [comment]
Hmm. EatThisMuch above puts 116 g of cubed pumpkin at 30 calories, whereas USDA has 49 calories to 245 g mashed pumpkin, which sounds like the ETM indicates an even higher caloric value than USDA does. Now, ETM's data does derive from the USDA entry on "raw pumpkin", as opposed to the transformed state of cooked/mashed, so if we leveled this against the C:DDA 525 g cut pumpkin, one cut pumpkin should be, what, 120+ calories? Not the whole pumpkin, but just one part.

Note that the links we're looking at, such as HowMuchIsIn, indicate a ratio of 1 lb cut pumpkin for 4 cups cubed for 1 cup mashed. If we just ignore mashed, that brings us close to a 525 g "cut pumpkin" comprising four 116 g cups cubed. A full game pumpkin with 4 cut pumpkin, or 16 cups pumpkin, would thus be 480 calories.

**Pumpkins ain't peanuts!!** 

Whatever the case, I think it's only right for us to sort it out here, otherwise we would end up with separate conflicting PRs; wouldn't that be a waste of time? 

-- Eager to Pump That Kin

**Comment by fairyarmadillo** [comment]
> Note that the links we're looking at, such as HowMuchIsIn, indicate a ratio of 1 lb cut pumpkin for 4 cups cubed for 1 cup mashed. If we just ignore mashed, that brings us close to a 525 g "cut pumpkin" comprising four 116 g cups cubed. A full game pumpkin with 4 cut pumpkin, or 16 cups pumpkin, would thus be 480 calories.

If pumpkins had that many calories, the pilgrims would not have had to figure out corn, which they wrote extensively about hating.

You're kinda veering all over the place with your sources, and comparing apples to oranges. A cup of cubed pumpkin cooks down into less than a cup of pumpkin, because it's mostly air and water. As it's cooked, the cell walls collapse and it shrinks down into mush, and water evaporates out of it. Remember also that the whole pumpkin contains a lot of inedible matter.

A sugar pie pumpkin contains maybe 160-200 kcal, tops - squashes are very low-energy food. We arbitrarily split that into four pieces, which are somewhat arbitratily 1 cup each, which even more arbitrarily each cook into 1 cup servings. This being a survival sim, the most important thing is that overall the item is not calorically overvalued.

**Comment by fairyarmadillo** [comment]
You're jumping around a lot, it's hard to follow and I think you're comparing apples to oranges in a few sources.

I also figured out what my error was.

Our pumpkin is a ~5lb sugar pumpkin, which looks about like this:

image

I frequently cook with these IRL. The yield I get when I remove the seeds, bake them, and scoop out and mash the flesh is about 2 cups. 

When I set their current values, I made some measurement errors - the USDA serving size for a pumpkin is 1/2 cup. There are 4 of these servings in one of these pumpkins.

The usual size for a serving was 1 cup. So I accidentally doubled the portion sizes. 1 pumpkin should only produce 500ml of pumpkin puree for a total of 180kcal, going by the Libby can saying it is 45 kcal per _half_ cup.

So 1 pumpkin should cut into 2 cut pumpkin, and each should be 90kcal.

**Comment by Montimorency** [comment]
Good point, but in my last comment there was new information to note. If we're comparing apples to apples - raw pumpkin to raw pumpkin, grams to grams - then we currently have a 2270 gram pumpkin that divides into 4 525 gram cuts, for a wastage of 170 grams (7.5%). 

Set aside mashed pumpkin for a moment. I somehow just realized mashed pumpkin is a separate, craftable item. We'll deal with it last, as the stats follow from the rest.

According to our various web sources, the skin/rind of the pumpkin is edible. Under apocalyptic conditions, waste not, want not? We did just add earthworms as a food resource... Perhaps that 170 g is just the stem. 

If we can treat the skin as edible, then our task is much simpler. All we have to do to begin with is adjust cut pumpkin nutrition to match calories, iron, calcium, and Vitamin C, proportionally to weight, according to this USDA data. Then the whole pumpkin nutrition follows, whether we decide we want 1:1 conservation in components or not. 
https://fdc.nal.usda.gov/fdc-app.html#/food-details/168448/nutrients

Can we do that? We would then have 26 kcal * 21 centigrams = 546 kcal. I don't see how that's unreasonable for an entire 5 lb pumpkin, minus stem. It's just that the current pumpkin in-game is outrageously under-nutritive. 

After that, if you insist, we can adjust the ingredients and output to the "mashed pumpkin" recipe so that a whole pumpkin would produce two portions of mash. (Actually, this recipe is currently broken, as it turns one 525 g cut pumpkin into 550 g of mashed pumpkin.)

**Comment by fairyarmadillo** [comment]
Why do you keep adding kcal every time you recalculate?

**Comment by Montimorency** [comment]
The increase was from applying nutritional values of raw pumpkin per se, and at that to a higher proportion of the total mass. 

OK, bottom line, you're both veterans here and I'm not going to get one over you: how much should the game pumpkins weigh, and how much weight in usable substance should they produce on disassembly? Would you be able and willing to dice and weigh a model pumpkin offline? 

From that point on, this link provides everything necessary. 
https://fdc.nal.usda.gov/fdc-app.html#/food-details/168448/nutrients

Though infrequently, I've bought both whole sugar pumpkins before, and wrapped slices like this giant hunk - this would be a large baking sheet's worth. I struggle to believe something like the image would contain fewer than 100 calories. It's definitely more than 500 grams of flesh alone, at least...

142137515-sliced-pumpkin-with-plastic-wrap-isolated-on-white-background

**Comment by fairyarmadillo** [comment]
...That is not a sugar pumpkin. I don't know what that is but it's got substantially more flesh than a sugar pumpkin. Sugar pumpkins are little and orange.

Let's do some math and see how much flesh there actually is in a sugar pumpkin.

A 5lb sugar pumpkin is, on average, 152mm across. source: (https://www.homedepot.com/p/Online-Orchards-2-Live-Pie-Pumpkin-Freshly-Harvested-Small-5-lbs-Size-Each-AUPK002/322607462)

The volume of a 152mm sphere is 1.84 liters. If we assume that the edible flesh of the pumpkin is 38mm thick, then the interior of the sphere is .77 liters, giving is 1.07 liters of flesh. 

image

That's a bit over 4 cups, but the flesh is not uniformly thick, and the stuff around the stem is woody and inedible. We'll call it 4 cups for convenience.

1 cups of cubed, cut up pumpkin is, per https://www.nutritionvalue.org/Pumpkin%2C_raw_nutritional_value.html 30 kcal. Times 4 is 120kcal, which is exactly what we have now.

Pumpkins do not have a lot of calories. They are mostly fiber, and you get surprisingly little edible food out of one.

**Comment by Montimorency** [comment]
The image is just for illustration, since I can't find what I have experience with. But I believe the wrapped slices I have reference to in stores might be Dickinson pumpkins. 

Pumpkin skin is not dense enough for it plus pulp plus stem plus seeds to constitute 80% of the pumpkin's weight, which is what you're attributing in your calculation. The skin on the pumpkin in the image above is very thin. What I'm contending is that the flesh in that 80 oz pumpkin, if we're simulating exactly the flesh, can't be at most a mere 17 oz. 

Hard figures of any kind remain difficult to identify online, but this p[aper](https://www.sciencedirect.com/science/article/pii/S2772753X22000557) contains a brief but useful mention: 

> A food industry in New Zealand named as Cedenco Foods Ltd produces 5500 metric tons of pumpkin waste during one processing season of 6 months. This waste material is almost 25% of total pumpkin processed because only flesh fraction is used for processing 

It is much more believable that 75% of a pumpkin in weight would be the flesh than 20%. Empty space has near-zero weight after all. If I had a food-grade scale I would attempt to validate this myself. It would be much better to resolve this by recourse to weight than to volume. 

Either way, the vitamins still need fixing, as does the mashed recipe.

**Comment by fairyarmadillo** [comment]
Industrial pumpkin farms producing fruit for canning don't use sugar pumpkins, they use much larger varieties which have a lower waste-to-flesh ratio. Our pumpkins are sugar pumpkins because those are the ones you find at grocery stores and are very commonly sold as garden vegetables because most people don't have room in their flowerbed for meter-wide pumpkins. There are no industrial pumpkin farms in the game and you most often find seeds in garden supply centers or homes.

You are right that the mashed pumpkin recipe and vitamins are all wrong. Mashed pumpkin is a super old item.

**Comment by Montimorency** [comment]
I am so confident that sugar pumpkins are not so radically divergent in mass composition that the only reason I'm not ordering a $10 scale on Amazon is that I would almost surely never use it again. Is there an alternative approach? A compromise on 50% flesh? 

Tangentially, you keep referring to a 120 calorie pumpkin - and this is indeed what HGtC and my local JSONs list - but in any world of mine, as mentioned, pumpkins are and have been 90 calories. Looking into it, it appears there's a general -25% calories modifier to raw comestibles in my local games. Huh.

**Comment by fairyarmadillo** [comment]
> I am so confident that sugar pumpkins are not so radically divergent in mass composition that the only reason I'm not ordering a $10 scale on Amazon is that I would almost surely never use it again. Is there an alternative approach? A compromise on 50% flesh?

It's up to Karol. I think we've both made our points here as well as can be expected, we don't want to muck up their comments with any more bikeshedding than we already have done.

> Tangentially, you keep referring to a 120 calorie pumpkin - and this is indeed what HGtC and my local JSONs list - but in any world of mine, as mentioned, pumpkins are and have been 90 calories. Looking into it, it appears there's a general -25% calories modifier to many raw comestibles in my local games. Huh.

Yes, foods with the RAW flag have their calorie values reduced unless they're cooked. The 120 kcal are there, potentially, but if you eat it raw you're only getting 90, as most uncooked food is not digested as well as cooked, even if it's technically edible.

**Comment by Montimorency** [comment]
I somehow didn't associate the pumpkins with that class, I was so focused on the topline. 

Videos like this already do a good job demonstrating the proportionality of the skin and other components. It's the weight that would really settle things. 

https://www.youtube.com/watch?v=WwwZdojA8fY

**Comment by fairyarmadillo** [comment]
We know the weight. She says in the video that she used two 3 pound sugar pumpkins and got five cups of puree. If a cup of pumpkin puree is 80kcal then she got 1.5 cups per pumpkin **which equals 120kcal per pumpkin.** Exactly what we have in game!

Admittedly, ours are slightly bigger than hers. It is not my experience that the flesh scales up 1:1, but even if it did, 120 * 1.6666 would be just about 200kcal for a 5 pound pumpkin.

**Comment by Montimorency** [comment]
No need to math, it's 5/6 vs. 4/5, or more or less the same ratios as in the links upthread (i.e. the ones that bring the whole pump calories to ~180).

**Comment by Karol1223** [comment]
Waking up in the morning to check your phone notifications only to see 14 new comments on GitHub and first thing you notice is a kukri embedded in a pumpkin was not exactly on my to-do list, believe it or not. I was not aware I had the capacity for these levels of confusion.

It is also a terrifying thought how long people can talk about pumpkins.

I'll be honest. You lost me halfway through. I stopped following what point you're trying to make around the second time the numbers changed. I don't even understand what's the final conclusion here.

I therefore am making an executive decision that I'm leaving pumpkins for last because no plant has the right to be this confusing.

**Comment by Montimorency** [comment]
Random update, but this individual found that a 30 oz pumpkin would produce equivalent to 15 oz of mashed pumpkin. While this is not 'cut to cut', it does offer more support to at least a 0.5 ratio for usable flesh to whole weight. 
https://ohsheglows.com/how-to-roast-a-sugar-pumpkin-make-fresh-pumpkin-puree-a-step-by-step-photo-tutorial

**Comment by Karol1223** [comment]
> ...That is not a sugar pumpkin. I don't know what that is but it's got substantially more flesh than a sugar pumpkin. Sugar pumpkins are little and have a smooth orange exterior.
> 
> Let's do some math and see how much flesh there actually is in a sugar pumpkin.
> 
> A 5lb sugar pumpkin is, on average, 152mm across. source: (https://www.homedepot.com/p/Online-Orchards-2-Live-Pie-Pumpkin-Freshly-Harvested-Small-5-lbs-Size-Each-AUPK002/322607462)
> 
> The volume of a 152mm sphere is, using 4/3πr^3, 1.84 liters. If we assume that the edible flesh of the pumpkin is 38mm thick, then the interior of the sphere is .77 liters, giving is 1.07 liters of flesh.
> 
> image The bottom pumpkin here has had the inedible pulp and the seeds removed, leaving only the rind and the edible flesh.
> 
> 1.07 liters is a bit over 4 cups, but the flesh is not uniformly thick, and the stuff around the stem is woody and inedible. We'll call it 4 cups for convenience.
> 
> 1 cups of cubed, cut up pumpkin is, per https://www.nutritionvalue.org/Pumpkin%2C_raw_nutritional_value.html 30 kcal. Times 4 is 120kcal, which is exactly what we have now. Or we could go with your usda.gov source and reduce it to 26kcal/cup for 104kcal total.
> 
> Pumpkins do not have a lot of calories. They are mostly fiber and water, and you get surprisingly little edible food out of one considering their size. It looks like the existing whole pumpkin item's volume, which was a rough estimate made by me, who went off of calories and weight and just guessed about volume, is too high, and should be reduced to 1.84L. Cut pumpkin should be 250ml. Weight is correct.

@fairyarmadillo apologies but there's one thing I'm not following here. The website you linked says a cup of pumpkin flesh is ~116g (so our cut pumpkin weight isn't right) - but then... we're losing like 1.8kg of mass when cutting the pumpkin up? Is that normal and expected?

**Comment by fairyarmadillo** [comment]
Cut pumpkin's mass was eyeballed and is probably wrong. The seeds, pulp, stem, and rind have been removed which is about 1/5 of the fruit by volume. Most of that stuff is heavier than the raw flesh, but I am sure I undershot the mass.

**Comment by Montimorency** [comment]
Check out [that video](https://www.youtube.com/watch?v=WwwZdojA8fY) in of the woman baking pumpkin and compare the slabs of pumpkin retrieved from the whole. Would you really call those a mere fifth or quarter of the whole by weight without concrete evidence? 

> I am pretty confident the proportions would vastly differ as the pumpkin diameter increases?

The growth in volume should be predominantly accounted for by flesh, in which case growth in weight/diameter of the object should favor flesh, not penalize it.  

I strongly recommend just using "raw pumpkin" -> 50% flesh as the starting point for nutrition and recipes. It's a good compromise.
